tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -744,5 +746,87 @@ public void TestJoin()
             Assert.Equal(join["Int_right"][2], right["Int"][2]);
             VerifyJoin(join, left, right, JoinAlgorithm.Inner);
         }
+
+        [Fact]
+        public void TestArrowIntegration()
+        {
+            Field fInt = new Field.Builder().Name("Int").DataType(Int32Type.Default).Build();
+            Field fFloat = new Field.Builder().Name("Float").DataType(FloatType.Default).Build();
+            Schema schema = new Schema.Builder().Field(fInt).Field(fFloat).Build();
+
+            var intArrayBuilder = new Int32Array.Builder().Append(5).Append(10).Append(5).Append(10);
+            Int32Array intArray = intArrayBuilder.Build();
+            var floatArrayBuilder = new FloatArray.Builder().Append(5.0f).Append(10.0f).Append(5.0f).Append(10.0f);
+            FloatArray floatArray = floatArrayBuilder.Build();
+            List<IArrowArray> arrowArrays = new List<IArrowArray>() { intArray, floatArray };
+            RecordBatch recordBatch = new RecordBatch(schema, arrowArrays, 4);
```

Comment:
(nit) you can use the RecordBatch.Builder API here. It would make this a little more concise.